src/sources/file_descriptors/stdin.rs

```diff
@@ -73,11 +81,36 @@ impl_generate_config_from_default!(StdinConfig);
 #[async_trait::async_trait]
 impl SourceConfig for StdinConfig {
     async fn build(&self, cx: SourceContext) -> crate::Result<crate::sources::Source> {
-        self.source(io::BufReader::new(io::stdin()), cx.shutdown, cx.out)
+        let log_namespace = cx.log_namespace(self.log_namespace);
+        self.source(
+            io::BufReader::new(io::stdin()),
+            cx.shutdown,
+            cx.out,
+            log_namespace,
+        )
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(self.decoding.output_type())]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        let host_key_path = self.host_key.as_ref().map_or_else(
+            || owned_value_path!(log_schema().host_key()),
+            |x| owned_value_path!(x),
+        );
+
+        let schema_definition = self
+            .decoding
+            .schema_definition(log_namespace)
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::InsertIfEmpty(host_key_path)),
+                &owned_value_path!("host"),
+                Kind::bytes(),
+                None,
+            )
+            .with_standard_vector_source_metadata();
+
+        vec![Output::default(self.decoding.output_type()).with_schema_definition(schema_definition)]
```

Comment:
Looks like this is the exact same as the `file_descriptors/file_descriptor.rs` one.

I think it'd be nice to extract that into `file_descriptors/mod.rs` , as the two are unlikely to diverge.

Reply:
I did a quick exercise to see how I would do this, with a new trait. I actually don't think it would be necessarily an improvement--we'd add one trait requiring 4 methods from a struct, then two implementations. This overhead seems pretty high for simplifying 19 lines of code. Any thoughts?

Reply:
Hmm I was thinking less along the lines of a trait and more just a helper function in the mod.rs.
Something like (off the cuff may not compiled as-is) :

```Rust

pub(crate) fn get_outputs(log_namespace: LogNamespace, decoding: DeserializerConfig, host_key: Option<&str>, source_name: &'static str) -> Vec<Output>  {
        let host_key_path = host_key.map_or_else(
            || owned_value_path!(log_schema().host_key()),
            |x| owned_value_path!(x),
        );

       let schema_definition = decoding
            .schema_definition(log_namespace)
            .with_source_metadata(
                source_name,
                Some(LegacyKey::InsertIfEmpty(host_key_path)),
                &owned_value_path!("host"),
                Kind::bytes(),
                None,
            )
            .with_standard_vector_source_metadata();
            
    vec![Output::default(decoding.output_type()).with_schema_definition(schema_definition)]
}

// used :

    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
        let log_namespace = global_log_namespace.merge(self.log_namespace);
        super::get_outputs(log_namespace, self.decoding, self.host_key.as_ref())
    }
```

Reply:
Ah gotcha, I'll try that! I might've been trying to force a trait 😅